src/main/java/use_cases/PostManager.java

```diff
@@ -0,0 +1,73 @@
+package use_cases;
+
+import entities.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+/*
+store all the posts existing in the app,
+add a new post, comment/like a Post
+ */
+public class PostManager {
+    private ArrayList<Post> posts;
+
+    public PostManager(){
+        this.posts = new ArrayList<Post>();
+    }
+    //in case
+    /*
+        Returns true if the interaction could be done, false otherwise
+        if the type is true, then the interaction is a comment
+     */
+    private boolean interactPost(Post post, Object obj, boolean type){
```

Comment:
Could we have two of these methods so we can remove the ambiguous "Object" variable and type boolean? I think we could use overloading like this:

private boolean interactPost(Post post, User user)
private boolean interactPost(Post post, Comment comment)

Reply:
I thought of that, but what it would be the same code, no?

Reply:
If you want to prevent repeated code, you can write a helper method that finds the Post and returns it. Then there will be no repeated code. I definitely feel like we should use the overloaded methods to prevent TypeErrors here.